src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,279 @@
+
+        return ResponseEntity.ok().body(tutorialGroup);
+    }
+
+    /**
+     * POST /courses/:courseId/tutorial-groups : creates a new tutorial group.
+     *
+     * @param courseId      the id of the course to which the tutorial group should be added
+     * @param tutorialGroup the tutorial group that should be created
+     * @return ResponseEntity with status 201 (Created) and in the body the new tutorial group
+     */
+    @PostMapping("/courses/{courseId}/tutorial-groups")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<TutorialGroup> create(@PathVariable Long courseId, @RequestBody @Valid TutorialGroup tutorialGroup) throws URISyntaxException {
+        log.debug("REST request to create TutorialGroup: {} in course: {}", tutorialGroup, courseId);
+        if (tutorialGroup.getId() != null) {
+            throw new BadRequestException("A new tutorial group cannot already have an ID");
+        }
+
+        var course = courseRepository.findByIdElseThrow(courseId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        tutorialGroup.setCourse(course);
+
+        trimStringFields(tutorialGroup);
+        checkTitleIsUnique(tutorialGroup);
+        TutorialGroup persistedTutorialGroup = tutorialGroupRepository.save(tutorialGroup);
+
+        return ResponseEntity.created(new URI("/api/tutorial-groups/" + persistedTutorialGroup.getId())).body(persistedTutorialGroup);
+    }
+
+    /**
+     * DELETE tutorial-groups/:tutorialGroupId : delete a tutorial group.
+     *
+     * @param tutorialGroupId the id of the tutorial group to delete
+     * @return the ResponseEntity with status 204 (NO_CONTENT)
+     */
+    @DeleteMapping("/tutorial-groups/{tutorialGroupId}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Void> delete(@PathVariable Long tutorialGroupId) {
+        log.info("REST request to delete a TutorialGroup : {}", tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+        tutorialGroupRepository.deleteById(tutorialGroupFromDatabase.getId());
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * PUT /tutorial-groups : Updates an existing tutorial group
+     *
+     * @param tutorialGroup group the tutorial group to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated tutorial group
+     */
+    @PutMapping("/tutorial-groups")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<TutorialGroup> update(@RequestBody @Valid TutorialGroup tutorialGroup) {
```

Comment:
I think you misunderstood me here.
I did not mean that you should remove the `/course/{course-id}` here, but that you should also include the `{tutorialGroupId}` in the URL, so it would be like this:
```java
@PutMapping("programming-exercises/{exerciseId}/exercise-hints/{exerciseHintId}")
    @PreAuthorize("hasRole('EDITOR')")
    public ResponseEntity<ExerciseHint> updateExerciseHint(@RequestBody ExerciseHint exerciseHint, @PathVariable Long exerciseHintId, @PathVariable Long exerciseId) {
```

Sorry if I wasn't clear enough here. The course part is correct, but you could also leave it out if you want (Though you should ask Stephan if he is fine with you leaving it out). You also currently aren't consistent in this file.

Reply:
@Hialus I left it in where I need the information. For example when creating a tutorial group I want to know to which course. I could get this also from the @RequestBody but I find it a little bit clearer.

I like this simplified way now where I only put the id in the URL I really need which means I do not have to pass around ids that much in the components in the client.

What is your opinion @krusche ?

Reply:
I agree with @Hialus, for a PUT request that updates a single entity you'd normally want to have the id as part of the URL.
And to make it consistent with other resources, we normally include the course id as well.

Reply:
I made all endpoints consistent in c926db4b405b0d47b989056509025ea28d1098c5